annif/project.py

```diff
@@ -117,6 +119,29 @@ def _suggest_with_backend(self, text, backend_params):
         logger.debug("Got %d hits from backend %s", len(hits), self.backend.backend_id)
         return hits
 
+    def _batched(self, iterable, n):
+        # From https://docs.python.org/3/library/itertools.html#itertools-recipes
+        it = iter(iterable)
+        while True:
+            batch = list(islice(it, n))
+            if not batch:
+                return
+            yield batch
+
+    def _suggest_batch_with_backend(self, corpus, backend_params):
+        if backend_params is None:
+            backend_params = {}
+        beparams = backend_params.get(self.backend.backend_id, {})
+
+        hit_sets = []
```

Comment:
Here `hit_sets` can potentially become quite large, for example in an `eval` operation on thousands of documents. Would it be possible to restructure this so that `_suggest_batch_with_backend` (or at least the outer `suggest_batch`) would return a generator instead? This way there would be no need to collect lots of hit_sets in a single list, but instead the calling code would start getting back results early, as soon as the first minibatch is finished. Perhaps [itertools.chain](https://docs.python.org/3/library/itertools.html#itertools.chain) or its sibling chain.from_iterable would be useful here?

Reply:
Easiest would be to use `yield self.backend.suggest_batch(...)` here in `_suggest_batch_with_backend`, and then use `return itertools.chain(self._suggest_batch_with_backend(...))` in the outer method `suggest_batch`.